xcrypto/hmac_test.go

```diff
@@ -75,6 +75,26 @@ func TestHMACUnsupportedHash(t *testing.T) {
 	}
 }
 
+func TestHMACAllocations(t *testing.T) {
+	h := xcrypto.NewHMAC(xcrypto.NewSHA256, nil)
+	msg := []byte("hello world")
+	n := int(testing.AllocsPerRun(10, func() {
+		h.Write(msg)
+		h.Sum(nil)
+		h.Reset()
+	}))
+
+	want := 2
+	if compareCurrentVersion("go1.24") >= 0 {
+		// The go1.24 compiler is able to optimize the allocation away.
+		// See cgo_go124.go for more information.
+		want = 1
```

Comment:
Same as previous comment.

Reply:
it is actually doing more allocation in stdlib HMAC.